code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/PasswordDaoImpl.java

```diff
@@ -22,29 +23,47 @@
 import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
 import io.github.ibuildthecloud.gdapi.util.TransformationService;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.util.Date;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import javax.inject.Inject;
 
 import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Form;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.entity.ContentType;
+
+import io.cattle.platform.json.JsonMapper;
+
+import com.netflix.config.DynamicStringProperty;
 
 public class PasswordDaoImpl extends AbstractJooqDao implements PasswordDao {
 
+    public static final DynamicStringProperty AUTH_VALIDATE_URL = ArchaiusUtil.getString("api.auth.local.validate.url");
+
     @Inject
     ObjectManager objectManager;
     @Inject
     GenericResourceDao resourceDao;
     @Inject
     TransformationService transformationService;
+    @Inject
+    JsonMapper jsonMapper;
 
     @Override
     public Credential changePassword(Credential password, ChangePassword changePassword, Policy policy) {
 
         if (policy.isOption(Policy.AUTHORIZED_FOR_ALL_ACCOUNTS) || transformationService.compare(changePassword.getOldSecret(), password.getSecretValue())){
             password.setSecretValue(transformationService.transform(changePassword.getNewSecret(), EncryptionConstants.HASH));
             password = objectManager.persist(password);
+            validatePassword(changePassword.getNewSecret());
```

Comment:
validatePassword() should be done before persisting the password to DB, also do the check in the caller instead of the Dao